cpp/src/arrow/filesystem/azurefs.cc

```diff
@@ -381,6 +388,24 @@ AzureOptions::MakeDataLakeServiceClient() const {
   return Status::Invalid("AzureOptions doesn't contain a valid auth configuration");
 }
 
+Result<std::string> AzureOptions::GenerateSASToken(
+    Storage::Sas::BlobSasBuilder* builder) const {
+  if (storage_shared_key_credential_) {
+    return builder->GenerateSasToken(*storage_shared_key_credential_);
+  } else {
+    // This part isn't tested. This may not work.
+    ARROW_ASSIGN_OR_RAISE(auto client, MakeBlobServiceClient());
```

Comment:
The caller already has a `BlobServiceClient`, so it's probably a good idea to have a version of `GenerateSASToken` that takes one as parameter.

Reply:
OK. I've added it.